src/utils/mtev_dyn_buffer.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2017, Circonus, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ *      notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ *      copyright notice, this list of conditions and the following
+ *      disclaimer in the documentation and/or other materials provided
+ *      with the distribution.
+ *    * Neither the name Circonus, Inc. nor the names of its contributors
+ *      may be used to endorse or promote products derived from this
+ *      software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef MTEV_DYN_BUFFER_H
+#define MTEV_DYN_BUFFER_H
+
+#include <mtev_defines.h>
+
+/* a struct that uses up 4K of stack space but then dynamically grows
+ * into heap space as you add to it
+ * 
+ * These are meant to be used on the stack ala:
+ * 
+ * mtev_dyn_buffer_t buf;
+ * mtev_dyn_buffer_init(&buf);
+ * mtev_dyn_buffer_add(&buf, "12345", 5);
+ * mtev_dyn_buffer_add(&buf, "67890", 5);
+ * printf((const char *)mtev_dyn_buffer_data(&buf)); // prints "1234567890"
+ * mtev_dyn_buffer_destroy(&buf);
+ */
+typedef struct mtev_dyn_buffer {
+  uint8_t static_buffer[4096];
+  size_t size;
+  uint8_t *data;
+  uint8_t *pos;
+} mtev_dyn_buffer_t;
+
+/*! \fn void mtev_dyn_buffer_init(mtev_dyn_buffer_t *buf)
+ *  \brief initialize a dyn_buffer
```

Comment:
I *think* the docs will not generate correctly with block *'s  It should be white-space up to the \ commands on all likes but the opening comment.